### ISSUE TITLE
Mutli-arch support

### ISSUE THREAD

**Issue by hendrix04** [kichik/docker-combo#5]
I noticed that currently you only build for the default amd64 platform. Curious as to if you would consider multi architecture support.

I am more than happy to make a PR to add the support in if it is something you would support going forward.

**Comment by kichik** [comment]
Yeah I would absolutely accept a PR for that. Sounds like a great idea.

**Comment by hendrix04** [comment]
Would you rather attempt to build every architecture for every combo or would you prefer to explicitly define a combo with it's architecture similar to "--override-from"?

The former really explodes how many builds need to run as well as requires explicit checks to ensure that each image supports that architecture.  The latter could make the combo list messy.

**Comment by kichik** [comment]
The more the merrier. If we can just add another variable to the build matrix and call it a day, that'd be awesome. We can then use exclusions to remove any images that aren't possible. But if the exclusion list will be huge, then probably the other way would be simpler.

**Comment by hendrix04** [comment]
My thought as well. Just wanted to confirm.

**Comment by hendrix04** [comment]
So mutli-arch breaks your testing routine. It doesn't look like the test does much other than makes sure that the container loads. I would expect the build to fail if the container wasn't going to load.

I can run the test AFTER the push, I can do a single arch build, test, then multi-arch build w/push, or skip the test all together. Your call.

**Comment by kichik** [comment]
The test also makes sure the installed tooling loads inside the container. Very basic, but better than nothing. 

How did you get it to build? With buildx? The documentation suggests that lets you also run arm64 on x86 as it uses qemu under the hood. Is that not the case in GitHub Actions?

**Comment by hendrix04** [comment]
Just made a PR so you can check it out for yourself!

I converted to using python-on-whales which allowed me to use buildx under the hood. I also had to add installing buildx and qemu to the build process. The good news is that it will now build for multiple architectures. The bad news is that it now takes like 4 hours to build.

While you can build a multiplatform image, you can't directly import one to your docker client. 
https://github.com/docker/buildx/issues/166
